application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -280,6 +326,66 @@ private boolean isWrongTagStatusAndHandle(@NotNull TagStatus requiredTagStatus,
         return false;
     }
 
+    private void logAction(@NotNull Subcommand subcommand, @NotNull Guild guild,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull String id,
+            @Nullable String newContent, @Nullable String previousContent) {
+
+        if (List
+            .of(Subcommand.CREATE, Subcommand.CREATE_WITH_MESSAGE, Subcommand.EDIT,
+                    Subcommand.EDIT_WITH_MESSAGE)
+            .contains(subcommand) && newContent == null) {
+            logger.debug("newContent is null even though the subcommand should supply a value.");
+            return;
+        }
+
+        if (List.of(Subcommand.EDIT, Subcommand.EDIT_WITH_MESSAGE, Subcommand.DELETE)
+            .contains(subcommand) && previousContent == null) {
+            logger
+                .debug("previousContent is null even though the subcommand should supply a value.");
+            return;
+        }
+
+        // to suppress warning "Argument '' might be null"
+        if (newContent == null) {
+            newContent = "";
+        }
+        if (previousContent == null) {
+            previousContent = "";
+        }
+
+        switch (subcommand) {
+            case CREATE -> ModAuditLogWriter.writeModAuditLog("Tag-Manage Create",
+                    String.format("created tag **%s**", id), author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment(CONTENT_FILE_NAME, newContent));
+
+            case CREATE_WITH_MESSAGE -> ModAuditLogWriter.writeModAuditLog(
+                    "Tag-Manage Create with message", String.format("created tag **%s**", id),
+                    author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment(CONTENT_FILE_NAME, newContent));
+
+            case EDIT -> ModAuditLogWriter.writeModAuditLog("Tag-Manage Edit",
+                    String.format("edited tag **%s**", id), author, timestamp, guild,
+                    List.of(new ModAuditLogWriter.Attachment(NEW_CONTENT_FILE_NAME, newContent),
+                            new ModAuditLogWriter.Attachment(PREVIOUS_CONTENT_FILE_NAME,
+                                    previousContent)));
+
+            case EDIT_WITH_MESSAGE -> ModAuditLogWriter.writeModAuditLog(
+                    "Tag-Manage Edit with message", String.format("edited tag **%s**", id), author,
+                    timestamp, guild,
+                    List.of(new ModAuditLogWriter.Attachment(NEW_CONTENT_FILE_NAME, newContent),
+                            new ModAuditLogWriter.Attachment(PREVIOUS_CONTENT_FILE_NAME,
+                                    previousContent)));
+
+            case DELETE -> ModAuditLogWriter.writeModAuditLog("Tag-Manage Delete",
+                    String.format("delete tag **%s**", id), author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment(PREVIOUS_CONTENT_FILE_NAME, previousContent));
```

Comment:
You should use the `actionVerb` from the enum in the text. I know that it makes the code more bloated but you have to be consistent. Someone who edits the `actionVerb` in `Subcommand` would expect it to change it here as well, so it has to be used here as well.

Reply:
done, 91f4238146a6e014b5466f4d1850e24bfd34bbfe